scipy/integrate/_py/base.py

```diff
@@ -159,17 +159,11 @@ def step(self):
                                "solver.")
 
         if self.n == 0 or self.t == self.t_crit:
-            # Handle corner cases of empty solver and no integration
```

Comment:
@drhagen Does this refactoring make sense?

I figured that a degenerate solver should be always 'finished' after a single step.